### ISSUE TITLE
Unable to install new packages

### ISSUE THREAD

**Issue by sbkl** [clerkinc/t3-turbo-and-clerk#46]
**Describe the bug**
When using `pnpm install <npm-package-name> --filter <app-name>`, the command line is frozen and nothing get installed.

**To Reproduce**
1. Go to the root of the monorepo
2. Run the following command for example: `pnpm install @headlessui/react --filter @acme/nextjs`
3. The command is frozen and nothing happened.

**Expected behavior**
Package being installed to the right repo filtered.

**Logs**
```
➜  t3-test git:(main) ✗ pnpm install @headlessui/react --filter @acme/nextjs
.                                        |  WARN  `node_modules` is present. Lockfile only installation will make it out-of-date
 WARN  Issues with peer dependencies found
apps/nextjs
└─┬ @clerk/nextjs 4.7.5
  └─┬ @clerk/clerk-react 4.8.3
    └─┬ @clerk/shared 0.9.1
      └── ✕ missing peer react-test-renderer@^18.2.0
Peer dependencies that should be installed:
  react-test-renderer@^18.2.0  

.                                        |    +1397   -3 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-
node_modules/core-js: Running postinstall script, done in 39ms
node_modules/core-js-pure: Running postinstall script, done in 58ms
node_modules/browser-tabs-lock: Running postinstall script, done in 40ms
node_modules/@prisma/engines: Running postinstall script, done in 199ms
node_modules/prisma: Running preinstall script, done in 50ms
node_modules/prisma: Running install script, done in 47ms
Progress: resolved 470, reused 453, downloaded 0, added 5, done
node_modules/turbo: Running postinstall script, done in 120ms
```

**Screenshots**
<img width="907" alt="Screenshot 2023-01-31 at 5 17 45 PM" src="https://user-images.githubusercontent.com/8094590/215733717-1684eebb-8066-493b-9556-f38c2d7d3956.png">

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.6
 - Browser Chrome
 - Version 109.0.5414.119 (Official Build) (arm64)

**Additional context:**
New to monorepo but by reading the docs from turbo-repo, this is the way described about how to add new package. Also, this is working properly with a brand new turborepo. I might be missing something like having the db migrated or other stuff. Will keep trying and see if I can get it working. In the meantime, appreciate if you can help with some direction.

**Comment by perkinsjr** [comment]
I can confirm installing a new dependency works as expected. Please try cleaning the workspaces and then installing it. 

I used the command provided and successfully installed it without any issues. 

![image](https://user-images.githubusercontent.com/45409975/215752628-0e0582ab-9d41-434a-83f3-7d3088bc786d.png)

**Comment by lukeberry99** [comment]
I'm running into the same issue with `pnpm i @headlessui/react --filter @acme/nextjs` 

![image](https://user-images.githubusercontent.com/1644147/215856849-989b9273-2a98-40f3-9b79-5dcf7cad92fc.png)


It works when using the t3-turbo repo as is, just not here for some reason.

**Comment by jjrise** [comment]
I've been having the same issue as well... a combination of deleting all the node_modules directories and the pnpm-lock.yaml file will eventually let me install/filter for a particular workspace.

**Comment by lukeberry99** [comment]
Downgrading to `pnpm 7.13.1` worked for me - I wonder if this is a pnpm issue with turbo 1.7.0

**Comment by lukeberry99** [comment]
This is really odd.

This is what I get with the t3-turbo repo

![image](https://user-images.githubusercontent.com/1644147/215859715-950456da-4f30-4ab7-bba6-aec3ebe45c4a.png)

And then this is what I get on this repo

![image](https://user-images.githubusercontent.com/1644147/215859334-d1ca43d3-10d2-492c-9a68-c929519e3019.png)


Both are freshly cloned, using node 18.13.0 and pnpm 7.26.2

I wonder if its related to the unmet peer dependency? But I doubt it. I'll keep digging, ty for taking a look @perkinsjr

**Comment by perkinsjr** [comment]
That unmet peer dependency doesn't affect anything. It's unused and was accidentally added during a PR for Clerk package. 

Just for my own sanity I switched machines again, to a windows machine, and then to an additional MacBook (that has used pnpm before), and cannot recreate this at all.

**Comment by jjrise** [comment]
I've had weird issues similar to this in the past that were related to using a M1 (arm) based macbook... are any of your macbooks newer m1/m2?

**Comment by perkinsjr** [comment]
> I've had weird issues similar to this in the past that were related to using a M1 (arm) based macbook... are any of your macbooks newer m1/m2?

One is an M1 Max and the other is a MBP Air (M1)

**Comment by perkinsjr** [comment]
> I don't know if these are related, but now I'm getting issues where the install doesn't complete fully... I'm doing a clean and then install, and inside my api package the only thing I'm seeing in node_modules is @acme/db. Is the normal behavior for turborepo that all the packages are installed in the root node_modules? Because right now all my packages are installing at the root/project level node_modules...
> 
> <img alt="image" width="345" src="https://user-images.githubusercontent.com/66971859/215864068-69b248c4-0215-4968-9840-50b523096dfc.png">
> 
> <img alt="image" width="675" src="https://user-images.githubusercontent.com/66971859/215864353-eee347a3-937f-41c9-a6d1-acd168bce49c.png">

This is similar to things we saw in the T3-Turbo repo in general where PNPM fails to resolve files. This usually requires you to either delete the PNPM lock. 

It isn't related to this per say.

**Comment by jjrise** [comment]
Gotcha. I have been deleting the lock file and installation is different, but still not correct. What was your "or" for what to do instead of deleting the lock file?

**Comment by perkinsjr** [comment]
You can try checking out a new template..  Or clearing out the PNPM store completely so it's forced to re download all files.

**Comment by jjrise** [comment]
yeah tried all of that... deleted the store from disk, started from a fresh template, and still having the issue. It has a different installation method depending on what version of PNPM I'm using.. 7.11.0 fails during install with ` Error: Cannot find module 'turbo'` and 7.13.1 finishes but all the packages are installed to the root node_modules and there isn't even node_modules in `app/nextjs`. 

I've been using this repo since your colab video with Theo and really haven't had any issues, so it's bizarre that it's randomly happening. Only major change was I installed React Query Dev Tools. But in all my recent attempts I removed that anyway.
